### ISSUE TITLE
bug: Live reload seems broken on fresh install of tools and totally new project

### ISSUE THREAD

**Issue by haroldpadilla** [ionic-team/ionic-framework#24407]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

Live reload doesn't work on `Google Chrome Version 96.0.4664.110 (Official Build) (x86_64)` on macOS. I started a fresh project using Ionic 6 with react and capacitor.

ionic cli version: 6.18.1
node: v17.2.0
npm: 8.1.4

The first time I run `ionic serve` it shows the app perfectly, but it breaks when I make a change. This is the console error in Chrome:

```
Uncaught ReferenceError: process is not defined
    at Object.4043 (<anonymous>:2:13168)
    at r (<anonymous>:2:306599)
    at Object.8048 (<anonymous>:2:9496)
    at r (<anonymous>:2:306599)
    at Object.8641 (<anonymous>:2:1379)
    at r (<anonymous>:2:306599)
    at <anonymous>:2:315627
    at <anonymous>:2:324225
    at <anonymous>:2:324229
    at HTMLIFrameElement.e.onload (index.js:1)
```

Live reload at least rerenders on `Firefox 95.0 (64-bit)` even though the same error shows in the console, but you can't click on anything. This is the error:

```
Uncaught ReferenceError: process is not defined
    4043 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    8048 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    8641 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    onload index.js:1
    be index.js:1
    he index.js:1
    tryDismissErrorOverlay webpackHotDevClient.js:184
    onHotUpdateSuccess webpackHotDevClient.js:109
    handleApplyUpdates webpackHotDevClient.js:257
    tryApplyUpdates webpackHotDevClient.js:273
    promise callback*tryApplyUpdates webpackHotDevClient.js:271
    handleSuccess webpackHotDevClient.js:106
    onmessage webpackHotDevClient.js:203
    js webpackHotDevClient.js:195
    Webpack 7
blank line 91312 > injectedScript:2:13158
    4043 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    8048 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    8641 blank line 91312 > injectedScript:2
    r blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    <anonymous> blank line 91312 > injectedScript:2
    onload index.js:1
    (Async: EventHandlerNonNull)
    be index.js:1
    he index.js:1
    tryDismissErrorOverlay webpackHotDevClient.js:184
    onHotUpdateSuccess webpackHotDevClient.js:109
    handleApplyUpdates webpackHotDevClient.js:257
    tryApplyUpdates webpackHotDevClient.js:273
    (Async: promise callback)
    tryApplyUpdates webpackHotDevClient.js:271
    handleSuccess webpackHotDevClient.js:106
    onmessage webpackHotDevClient.js:203
    (Async: EventHandlerNonNull)
    js webpackHotDevClient.js:195
    Webpack 7
        __webpack_require__
        fn
        1
        __webpack_require__
        checkDeferredModules
        webpackJsonpCallback
        <anonymous>
```

### Expected Behavior

Live Reload should work in any supported browser.

### Steps to Reproduce

Steps:

1. $ npm install -g @ionic/cli native-run cordova-res
2. $ ionic start photo-gallery tabs --type=react --capacitor
3. $ cd photo-gallery
4. $ ionic serve
5. You will be able to see the app render initially
5. Change a file and wait for the page reload
6. At this point your chrome will basically won't show you the app and there will be an error in the console
7. If you wish to restart the page, you could and this would show you the updated app

### Ionic Info

ionic cli version: 6.18.1
"@ionic/react": "^6.0.0",
"@ionic/react-router": "^6.0.0",

**Comment by haroldpadilla** [comment]
This issue can be fixed by updating react-scripts from 4.0.2 to 5.0.0 in your package.json (rerun npm install).

**Comment by liamdebeasi** [comment]
Thanks for the issue. The errors appear to be related to webpack and hot reloading, not Ionic.

Since the issue can be resolved by updating to the latest version of react-scripts (which also updated webpack), I am going to close this issue out. Thank you!

**Comment by Fosol** [comment]
This doesn't appear to be fixed in 5.0.0.  

I'm seeing the same issue.  On a hot reload the error is thrown and the whole page becomes unresponsive.  You have to refresh the page to get it working again.

Fails in all browsers.

```
Uncaught ReferenceError: process is not defined
    4043 blank line 66162 > injectedScript:2
    r blank line 66162 > injectedScript:2
    8048 blank line 66162 > injectedScript:2
    r blank line 66162 > injectedScript:2
    8641 blank line 66162 > injectedScript:2
    r blank line 66162 > injectedScript:2
    <anonymous> blank line 66162 > injectedScript:2
    <anonymous> blank line 66162 > injectedScript:2
    <anonymous> blank line 66162 > injectedScript:2
    onload index.js:1
    be index.js:1
    he index.js:1
    tryDismissErrorOverlay webpackHotDevClient.js:184
    onHotUpdateSuccess webpackHotDevClient.js:109
    handleApplyUpdates webpackHotDevClient.js:257
    tryApplyUpdates webpackHotDevClient.js:273
    promise callback*tryApplyUpdates webpackHotDevClient.js:271
    handleSuccess webpackHotDevClient.js:106
    onmessage webpackHotDevClient.js:203
    js webpackHotDevClient.js:195
    Webpack 7
blank line 66
```
